src/app

```diff
@@ -190,23 +213,60 @@ handle_artifact() {
-    $app_sub_module ROLLOUT "${application_name}" "${PERSISTENT_STORE}/${application_name}/manifests"
-    # now if $app_sub_module healthy then rm -Rf "${PERSISTENT_STORE}/${artifact_name}-${rollback_id}"
-    # and on rollback we just deploy "${PERSISTENT_STORE}/${artifact_name}-${rollback_id}"
-    # earlier ln -s "${PERSISTENT_STORE}/${artifact_name}-${rollback_id}" "${PERSISTENT_STORE}/${artifact_name}-rollback"
-    # and removing all else
+    echo "rolling out ${PERSISTENT_STORE}/${application_name}/manifests"
+    set +e
+    $app_sub_module ROLLOUT "${application_name}" "${PERSISTENT_STORE}/${application_name}/manifests" 2>&1
+    if test $? -eq 0; then
+        echo "successful rollout"
+        if test -d "${PERSISTENT_STORE}/${application_name}-${rollback_id}"; then
+            echo "cleaning up -${rollback_id}"
+            clean_up "${PERSISTENT_STORE}/${application_name}-${rollback_id}" # clean up call automatically removes the images in by reference taken from clen-up subdirectory, it relies on the fact that we save them there
+            echo "cleaning up -previous"
+            clean_up "${PERSISTENT_STORE}/${application_name}-previous" # clean up the composition that was running just now
+        else
+            echo "successful rollout: nothing to clean"
+        fi
+        rm -Rfv "${PERSISTENT_STORE}/${application_name}-previous"
+        # save_rollback "${PERSISTENT_STORE}/${application_name}-${rollback_id}" # saves the current images ids and manifests as new rollback state. this can be just cp -a "${PERSISTENT_STORE}/${application_name} "${PERSISTENT_STORE}/${application_name}-${rollback_id}"
+        echo "saving data for rollback"
+        rm -Rfv "${PERSISTENT_STORE}/${application_name}-${rollback_id}"
+        cp -va "${PERSISTENT_STORE}/${application_name}" "${PERSISTENT_STORE}/${application_name}-${rollback_id}"
+        echo "successfully saved rollback data"
+    else
+        echo "unsuccessful rollout"
+        rc=8
+        clean_up "${PERSISTENT_STORE}/${application_name}"
+        $app_sub_module ROLLBACK "${application_name}" "${PERSISTENT_STORE}/${application_name}-${rollback_id}" "${PERSISTENT_STORE}/${application_name}/manifests"
+        if test $? -eq 0; then
+            echo "successful rollback"
+            rc=80
+        else
+            echo "unsuccessful rollback trying to rollout the composition we saw when we started"
+            $app_sub_module ROLLOUT "${application_name}" "${PERSISTENT_STORE}/${application_name}-previous/manifests"
+            if test $? -eq 0; then
+                echo "successfully rolled out ${PERSISTENT_STORE}/${application_name}-previous"
+                rc=82
+            else
+                echo "rollout of ${PERSISTENT_STORE}/${application_name}-previous was unsuccessful; deployment failed; rollout attempts failed."
+                rc=84
+            fi
+        fi
+    fi
+    set -e
+    return $rc
```

Comment:
@tranchitella what I wonder is: should we support `SupportsRollback` below. I hope you can give me some advice once you review this apporach.

Reply:
Actually, I believe we don't need the "SupportsRollback" as it is built-in in the installation.
Therefore, I'd reject MEN-6077.

Reply:
thank you so what I did: I rejected https://northerntech.atlassian.net/browse/MEN-6197 since it was part of this change anyway. and I marked both MEN-6077 and MEN-6078 as done. the ROLLBACK as such was implemented here, so it basically makes sense to mark them as done. if you want me to reject it, I can do it.